core/src/main/java/nl/nn/adapterframework/pipes/WsdlXmlValidator.java

```diff
@@ -197,8 +178,40 @@ public void configure() throws ConfigurationException {
 		super.configure();
 	}
 
+	@Override
+	protected PipeForward validate(Message messageToValidate, PipeLineSession session, boolean responseMode, String messageRoot) throws XmlValidatorException, PipeRunException, ConfigurationException {
+		String soapAction = session.get(SoapBindingConstants.SOAP_ACTION, "");
+		if(StringUtils.isNotEmpty(soapAction)) {
+			setSoapBody(getSoapBodyFromSoapAction(soapAction));
+			resetCalculatedRootValidatons();
+			configure();
```

Comment:
Is the soapBody always set from soapAction? Could there be cases where the calculated root validations should not be reset?